UFile.go

```diff
@@ -0,0 +1,276 @@
+package ucloud
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+func (u *UcloudApiClient) SignatureUFile(privateKey string, stringToSign string) string {
+	mac := hmac.New(sha1.New, []byte(privateKey))
+	mac.Write([]byte(stringToSign))
+	return base64.StdEncoding.EncodeToString(mac.Sum(nil))
+}
+
+type SignParam struct {
+	HttpVerb                   string
+	ContentMd5                 string
+	ContentType                string
+	Date                       string
+	CanonicalizedUCloudHeaders string
+	CanonicalizedResource      string
+}
+
+func (param *SignParam) String() string {
```

Comment:
why? because the field is public or the SignatureUFile should be private ?

On Fri, May 6, 2016 at 3:39 PM, Meng Zhuo <notifications@github.com> wrote:

> In UFile.go
> <https://github.com/mengzhuo/ucloud-go-sdk/pull/3#discussion_r62297563>:
>
> > +func (u *UcloudApiClient) SignatureUFile(privateKey string, stringToSign string) string {
> > +	mac := hmac.New(sha1.New, []byte(privateKey))
> > +	mac.Write([]byte(stringToSign))
> > +	return base64.StdEncoding.EncodeToString(mac.Sum(nil))
> > +}
> > +
> > +type SignParam struct {
> > +	HttpVerb                   string
> > +	ContentMd5                 string
> > +	ContentType                string
> > +	Date                       string
> > +	CanonicalizedUCloudHeaders string
> > +	CanonicalizedResource      string
> > +}
> > +
> > +func (param *SignParam) String() string {
>
> This is unacceptable.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mengzhuo/ucloud-go-sdk/pull/3/files/11ac4a16928d228ae9604e3a1dfbfcdbaa1c49e2#r62297563>
>



-- 
夏海明